python/adjoint/optimization_problem.py

```diff
@@ -31,7 +31,8 @@ def __init__(
         decay_by=1e-11,
         decimation_factor=0,
         minimum_run_time=0,
-        maximum_run_time=None
+        maximum_run_time=None,
+        finite_difference_step=1e-6
```

Comment:
Shouldn't this step size used to compute the finite-difference term Aᵤ during the recombination step be dependent on the floating-point precision of the fields similar to the step size of the finite-difference gradient? A hard-coded value of `1e-6` could be susceptible to rounding errors for single-precision fields.

Reply:
Does the single-precision flag also force the permittivity to single precision? I thought only the fields were single precision. If my assumption is correct, then the finite difference term behaves the same in both configurations.

Reply:
Note that the `chi1inv` member of the `structure_chunk` class has type `realnum`:

https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/meep.hpp#L583

Hence, along with the fields of the `fields_chunk` class the permittivity is also compiled using single-precision floating point.

Reply:
True, but `eff_chi1inv_row` (which is what the recombination routine uses) actually asumes that `chi1inv_row` is always a `double`:

https://github.com/NanoComp/meep/blob/bc58782c0d16e6d59937028cbe2ffa402f5f24b0/src/meepgeom.cpp#L1014-L1015

Either way, the step size doesn't need to be so small. I reduced it to be safe.